### PR TITLE
For autogain runs, add voltage plotting scheme

### LINKED ISSUE #445: Notebook for extracting detector data for autogain testing.
**Is your feature request related to a problem? Please describe.**
Need a notebook to quickly pull out detector voltage info into a) a csv and b) a plot

**Describe the solution you'd like**
I have a notebook `detector_v_plotter` and a src file `fov_detector_calc` that provides this already - just want to integrate this into toffy with the appropriate tests.

**Additional context**
Example csv and plot below.
[detector_gain_values.csv](https://github.com/angelolab/toffy/files/13127359/detector_gain_values.csv)
detector_gain_plot

### PR DESCRIPTION
**Pull request by alex-l-kong**
**What is the purpose of this PR?**

Closes #445. Runs generated with autogain will have varying voltages per run, so `check_detector_voltage` should not run. Instead, the user should see a plot showing the voltage over acquisition order.

**How did you implement your changes**

A few changes will need to be made:

1. Add a watcher callback that generates this plot in real-time, in this way the user can track the autogain. This callback should be optional and not be specified if a notebook-defined `autogain` is not set
2. Provide a separate notebook that generates this plot post-run (similar to post-3a notebooks)
3. Remove the `check_detector_voltage` call if the `autogain` flag is set in the normalization notebook

### CODE REVIEW

**Comment by alex-l-kong** [comment]
We've decided to update this PR to only include the plotting functionality inside the 4b normalization notebook. An `autogain` flag can be set to `True` to generate this plot as opposed to verifying that the voltage was consistent across the run.

File: src/toffy/normalize.py [hunk 1]
```diff
@@ -892,3 +896,87 @@ def check_detector_voltage(run_dir):
...
+    # Get all the JSON run files
+    json_files = sorted(glob.glob(os.path.join(run_folder, "fov-*-scan-?.json")))
+
+    # Map each FOV to their observed detector voltage
+    detector_voltage_data = {"FOVs": [], "Detector Voltage": []}
+
+    # Iterate over the list of files
+    for json_file in json_files:
+        # Open the JSON file and load the data
+        with open(json_file, "r") as f:
```
Comment (camisowers) [inline src/toffy/normalize.py:931]: Could use json_utils.read_json_file() https://github.com/angelolab/toffy/blob/bdd0d6494e827c3f3f0bfb1fab076f02a4c259a3/src/toffy/json_utils.py#L96

**Review by camisowers — CHANGES_REQUESTED** [review]
Minor comments about using existing toffy functions.

**Review by srivarra — CHANGES_REQUESTED** [review]
Code looks good! What does this plot look like?

**Comment by alex-l-kong** [comment]
> Code looks good! What does this plot look like?

The linked issue (#445) contains an example.

**Review by srivarra — APPROVED** [review]

**Review by camisowers — APPROVED** [review]
Nice!

**Comment by bryjcannon** [comment]
From what I can tell things look good. 
The mibicontrol 1.8 warning is still present in the 4b notebook. You can remove this now, correct? Unsure if this is dependent on the other open PR dealing with autogain.

**Comment by alex-l-kong** [comment]
> From what I can tell things look good. The mibicontrol 1.8 warning is still present in the 4b notebook. You can remove this now, correct? Unsure if this is dependent on the other open PR dealing with autogain.

Yeah I can remove it before merging. My thinking was this PR is going to be merged in first before the actual autogain support PR, so _technically_ full autogain support won't be there yet. But agree it's kind of confusing, let's nuke.

**Review by bryjcannon — APPROVED** [review]
Great job on this, and thanks!

**Pull request merged**